fix(agw): Fixed compilation error on inclusion of proto_map.h file

rsarwad opened the pull request:
## Summary
fix(agw): Fixed compilation error on inclusion of proto_map.h file  and removed unused file, sgw.h and unused structure, predefined_pcc_rules 

## Test Plan
Executed s1ap sanity test suite and unit test cases

ShubhamTatvamasi committed:
docs(orc8r): updated slack link on the docs (#13745)

LKreutzer committed:
fix(ci): Fix debug logging for bazel dependency (#13747)

Signed-off-by: Lars Kreutzer <lars.kreutzer@tngtech.com>

Signed-off-by: Lars Kreutzer <lars.kreutzer@tngtech.com>

nstng committed:
fix: bazel base uses same folly version as magma vm (#13743)

Signed-off-by: Nils Semmelrock <nils.semmelrock@tngtech.com>

Signed-off-by: Nils Semmelrock <nils.semmelrock@tngtech.com>

nstng committed:
Revert "fix: bazel base uses same folly version as magma vm (#13743)" (#13753)

This reverts commit d08d78f495c7e304ce362372f7a9b889c6cfdb92.

Signed-off-by: Nils Semmelrock <nils.semmelrock@tngtech.com>

Signed-off-by: Nils Semmelrock <nils.semmelrock@tngtech.com>

jkmar committed:
chore(orc8r): Add option to enable http metrics for orc8r services (#13736)

Metrics are provided by echo framework Prometheus extension

Signed-off-by: Kuba Marciniszyn <kuba@freedomfi.com>

Signed-off-by: Kuba Marciniszyn <kuba@freedomfi.com>

jkmar committed:
feat(dp): Add relinquish custom action (#13715)

Signed-off-by: Kuba Marciniszyn <kuba@freedomfi.com>

Signed-off-by: Kuba Marciniszyn <kuba@freedomfi.com>

WojSad committed:
fix(dp): fix unique constraint error handling in sqlite (#13752)

Signed-off-by: Wojciech Sadowy <wojciech.sadowy@freedomfi.com>

Signed-off-by: Wojciech Sadowy <wojciech.sadowy@freedomfi.com>
Co-authored-by: Wojciech Sadowy <wojciech.sadowy@freedomfi.com>

rsarwad committed:
Modified code to fix compilation errors

Signed-off-by: Rashmi <rashmi.sarwad@radisys.com>

sreedharkumartn committed:
docs: Extended features in 5G SA documentation (#13624)

* docs: Extended features in 5G SA documentation

Signed-off-by: sreedharkumartn <sreedhar.kumar@wavelabs.ai>

* Addressed Review Comments (R1)

Signed-off-by: sreedharkumartn <sreedhar.kumar@wavelabs.ai>

* Review (R2)

Signed-off-by: sreedharkumartn <sreedhar.kumar@wavelabs.ai>

Signed-off-by: sreedharkumartn <sreedhar.kumar@wavelabs.ai>

Sathyaj27 committed:
fix(amf): ICS IE not required for ICS (#13318)

Signed-off-by: Sathyaj27 <sathya.jayadev@wavelabs.ai>

Signed-off-by: Sathyaj27 <sathya.jayadev@wavelabs.ai>

nstng committed:
chore: bazel base image is pinned so that new image can be released as a non breaking change (#13757)

Signed-off-by: Nils Semmelrock <nils.semmelrock@tngtech.com>

Signed-off-by: Nils Semmelrock <nils.semmelrock@tngtech.com>

rsarwad committed:
Removed files and structures that are no more required
Signed-off-by: Rashmi <rashmi.sarwad@radisys.com>

rsarwad committed:
Removed files and structures that are no more required
Signed-off-by: Rashmi <rashmi.sarwad@radisys.com>

ShubhamTatvamasi committed:
docs(cloud): updated pdf links in docs (#13759)

voisey committed:
feat(agw): Run integ tests versus containerized AGW (#13659)

Signed-off-by: Cameron Voisey <cameron.voisey@tngtech.com>

Signed-off-by: Cameron Voisey <cameron.voisey@tngtech.com>

Commit:
refactor(agw): remove duplicated implementations of DataModel methods (#13564)

Signed-off-by: Jarosław Jaszczuk <jaroslaw@freedomfi.com>

Signed-off-by: Jarosław Jaszczuk <jaroslaw@freedomfi.com>

LKreutzer committed:
chore(ci): Reconfigure bazel workflow trigger (#13751)

Signed-off-by: Lars Kreutzer <lars.kreutzer@tngtech.com>

Signed-off-by: Lars Kreutzer <lars.kreutzer@tngtech.com>

ShubhamTatvamasi committed:
docs(cloud): updated blog link (#13763)

Signed-off-by: Shubham Tatvamasi <shubhamtatvamasi@gmail.com>

Signed-off-by: Shubham Tatvamasi <shubhamtatvamasi@gmail.com>

sklgromek committed:
chore(dp): Remove apt packages version pinning in DP Dockerfiles (#13769)

Signed-off-by: Tomasz Gromowski <tomasz@freedomfi.com>

Signed-off-by: Tomasz Gromowski <tomasz@freedomfi.com>

LKreutzer committed:
chore(ci): Mitigate Vagrant cloud rate limiting errors by logging in (#13734)

Co-authored-by: Lars Kreutzer lars.kreutzer@tngtech.com
Co-authored-by: Krisztián Varga krisztian.varga@tngtech.com

Signed-off-by: Lars Kreutzer <lars.kreutzer@tngtech.com>

Signed-off-by: Lars Kreutzer <lars.kreutzer@tngtech.com>

MoritzThomasHuebner committed:
chore(agw): Python dependency upgrades (#13482)

* chore(agw): Updating python dependencies

- Upgraded versions in `lte/gateway/setup.py` and `orc8r/gateway/setup.py` where possible.
- Added `--python-pip /home/vagrant/build/python/bin/pip` to the `fpm` command in `pydep`. This option ensures that we are not using `easy_install` inadvertently.
- Updated the `magma.lockfile.ubuntu` by running `fab release package`.
- Added pylint exception. The linter erroneously flags this line otherwise.
- Updated bazel files.
- Also replaced `systemd` with `systemd-python` in bazel files since these are different packages

* bazel: downgrade pyyaml and generate requirements.txt changes

* chore(agw): Replace systemd with systemd_python

This change makes bazel consistent with the setup.py files. The two packages have similar interfaces but are different and cannot just be interchanged. The use of `systemd.daemon.notify` in `sdwatchdog.py` also indicates that `systemd_python` is the intended package one should use.

Signed-off-by: Moritz Huebner <moritz.huebner@tngtech.com>
Signed-off-by: Alexander zur Bonsen <alexander.zur.bonsen@tngtech.com>
Co-authored-by: Alexander zur Bonsen <alexander.zur.bonsen@tngtech.com>

Commit:
chore(agw): remove core capturing logic all together as the host logic just works. In /var/cores you will find the captured core files (#13738)

Signed-off-by: Christian Krämer <christian.kraemer@tngtech.com>

Signed-off-by: Christian Krämer <christian.kraemer@tngtech.com>

vktng committed:
chore: Add improvements to the bazel sudo wrapper script (#13744)

* chore: Add "retry on failure" feature to Bazel sudo test wrapper script
Co-authored-by: Lars Kreutzer lars.kreutzer@tngtech.com
Co-authored-by: Krisztián Varga krisztian.varga@tngtech.com

Signed-off-by: Krisztián Varga <krisztian.varga@tngtech.com>

* chore: Add xml report generation to Bazel sudo test wrapper script
Co-authored-by: Lars Kreutzer lars.kreutzer@tngtech.com
Co-authored-by: Krisztián Varga krisztian.varga@tngtech.com

Signed-off-by: Krisztián Varga <krisztian.varga@tngtech.com>

Signed-off-by: Krisztián Varga <krisztian.varga@tngtech.com>

lucasgonze committed:
chore: add security acknowledgements (#13608)

In README.md, create "Hall of Fame" to acknowledge people who submit vulnerabilities.

In SECURITY.md, add instructions for maintaining acknowledgements. Refactor markdown for maintainability.

Clean extra blank lines per reviewer request https://github.com/magma/magma/pull/13608#discussion_r951351672. Clean list formatting per https://github.com/magma/magma/pull/13608#discussion_r951357256. Add fullstop per https://github.com/magma/magma/pull/13608#discussion_r951356526.

Signed-off-by: Lucas Gonze <lucas@gonze.com>

LKreutzer committed:
chore(ci): Run the sudo tests with bazel (#13772)

Co-authored-by: Lars Kreutzer lars.kreutzer@tngtech.com
Co-authored-by: Krisztián Varga krisztian.varga@tngtech.com

Signed-off-by: Lars Kreutzer <lars.kreutzer@tngtech.com>

Signed-off-by: Lars Kreutzer <lars.kreutzer@tngtech.com>

maxhbr committed:
chore(orc8r): upload vagrant box with vagrant cli (#13555)

* chore(orc8r): upload vagrant box with vagrant cli

Signed-off-by: Maximilian Huber <maximilian.huber@tngtech.com>

* chore(orc8r): Tidied vagrant cli usage

- Moved `--release` option to the beginning of `vagrant cloud publish`, suggested by `vagrant cloud publish --help`
- Added a missing backslash
- Tested script on personal vagrant account

Signed-off-by: Maximilian Huber <maximilian.huber@tngtech.com>
Signed-off-by: Moritz Huebner <moritz.huebner@tngtech.com>
Co-authored-by: Moritz Huebner <moritz.huebner@tngtech.com>

HannaFar committed:
refactor(nms): Migrate Alerts Receivers detail to new style (#13774)

Signed-off-by: HannaFar <hannafarag159@gmail.com>

Signed-off-by: HannaFar <hannafarag159@gmail.com>

vktng committed:
chore: Fix setup and teardown in Bazel integtest wrapper script (#13789)

Signed-off-by: Krisztián Varga <krisztian.varga@tngtech.com>

Signed-off-by: Krisztián Varga <krisztian.varga@tngtech.com>

sebathomas committed:
fix(agw): Downgrade sentry-sdk, grpcio and spyne (#13795)

The Python dependency upgrades generated a broken magma debian
package with higher dependencies that could not be fullfilled.

Additionally, the new version of spyne breaks the CI due to
permission issues.

Signed-off-by: Sebastian Thomas <sebastian.thomas@tngtech.com>

ssanadhya commented on lte/gateway/c/core/oai/test/spgw_task/spgw_procedures_test_fixture.cpp:
```diff
@@ -24,6 +24,7 @@ extern "C" {
 #include "lte/gateway/c/core/oai/include/gx_messages_types.h"
 #include "lte/gateway/c/core/oai/include/ip_forward_messages_types.h"
 #include "lte/gateway/c/core/oai/lib/3gpp/3gpp_23.401.h"
+#include "lte/gateway/c/core/oai/tasks/sgw/sgw_defs.hpp"
```
This should be outside `extern "C"` block

ssanadhya commented on lte/gateway/c/core/oai/tasks/sgw/spgw_state_converter.cpp:
```diff
@@ -16,11 +16,11 @@
  */
 
 #include "lte/gateway/c/core/oai/tasks/sgw/spgw_state_converter.hpp"
+#include "lte/gateway/c/core/oai/include/sgw_context_manager.hpp"
```
Move this after the `extern "C"` block as per [style guide](https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes)

ssanadhya commented on lte/gateway/c/core/oai/include/sgw_context_manager.hpp:
```diff
@@ -23,15 +23,18 @@
 
+#include <stdint.h>
+
+#include "lte/gateway/c/core/oai/include/spgw_state.hpp"
```
This should be after the `extern "C"` block as per the [style guide](https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes)

ssanadhya commented on lte/gateway/c/core/oai/include/sgw_s8_state.hpp:
```diff
@@ -13,14 +13,18 @@ limitations under the License.
 
+#include <pthread.h>
+
+#include "lte/gateway/c/core/oai/include/spgw_types.hpp"
```
This should be after the `extern "C"` block as per the [style guide](https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes)

ssanadhya commented on lte/gateway/c/core/oai/include/spgw_state.hpp:
```diff
@@ -17,16 +17,16 @@
 
+#include <pthread.h>
+
+#include "lte/gateway/c/core/oai/include/spgw_types.hpp"
```
This should be after the `extern "C"` block as per the [style guide](https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes)

ssanadhya commented on lte/gateway/c/core/oai/include/spgw_types.hpp:
```diff
@@ -30,10 +30,17 @@
 #ifndef FILE_SPGW_TYPES_SEEN
 #define FILE_SPGW_TYPES_SEEN
```
Please replace with `#pragma once`

ssanadhya commented on lte/gateway/c/core/oai/lib/mobility_client/MobilityClientAPI.hpp:
```diff
@@ -20,13 +20,14 @@
 
 #include "stdint.h"
 
+#include "lte/gateway/c/core/oai/include/spgw_state.hpp"
```
This should be after the C headers as per the [style guide](https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes)

ssanadhya commented on lte/gateway/c/core/oai/lib/pcef/pcef_handlers.cpp:
```diff
@@ -15,29 +15,30 @@
+#include "lte/gateway/c/core/oai/lib/pcef/PCEFClient.hpp"
+#include "lte/gateway/c/core/oai/lib/mobility_client/MobilityClientAPI.hpp"
+#include "lte/gateway/c/core/oai/include/spgw_types.hpp"
+
```
This should be after the `extern "C"` block as per the [style guide](https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes)

ssanadhya reviewed (changes requested):
Please review the order of includes in [style guide]((https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes)

pruthvihebbani committed:
fix(agw): c++ migration of nas/emm/msg files (#13754)

* Renamed files
* Resolved compilation errors

Signed-off-by: Pruthvi Hebbani <pruthvi.hebbani@radisys.com>

rsarwad committed:
Addressed review comments

Signed-off-by: Rashmi <rashmi.sarwad@radisys.com>

Commit:
refactor(enodebd): package code for enodebd devices (#13581)

* refactor(agw): package code for enodebd devices

Signed-off-by: Jarosław Jaszczuk <jaroslaw@freedomfi.com>

* resolve conflicts

Signed-off-by: Jarosław Jaszczuk <jaroslaw@freedomfi.com>

Signed-off-by: Jarosław Jaszczuk <jaroslaw@freedomfi.com>

ShubhamTatvamasi committed:
fix(agw): updated env.j2 file on master branch (#13742)

MoritzThomasHuebner committed:
fix(agw): Fix reinstall logic and error handling in pydep (#13799)

- Changed the pydep script to reinstall all packages with apt
  since already installed packages might not be registered with
  python. This solves an issue in the build process where the
  uninstallation of "idna" breaks some Python scripts.
- Fixed the error handling of fabric (catching exceptions does
  not work) so the script can run even if no new dependencies
  are built.

Signed-off-by: Moritz Huebner <moritz.huebner@tngtech.com>
Signed-off-by: Sebastian Thomas <sebastian.thomas@tngtech.com>
Co-authored-by: Sebastian Thomas <sebastian.thomas@tngtech.com>

LKreutzer committed:
chore(agw): Clean up unnecessary Make commands (#13782)

* chore(agw): Clean up unnecessary Make commands: build_c and test_c

Signed-off-by: Lars Kreutzer <lars.kreutzer@tngtech.com>

* chore(agw): Clean up unnecessary Make commands: clang_tidy_oai and clang_tidy_oai_upload

Signed-off-by: Lars Kreutzer <lars.kreutzer@tngtech.com>

* chore(agw): Clean up unnecessary Make commands: build_oai_clang and clang_warning_oai_upload

Signed-off-by: Lars Kreutzer <lars.kreutzer@tngtech.com>

* chore(agw): Clean up unnecessary Make commands: help

Signed-off-by: Lars Kreutzer <lars.kreutzer@tngtech.com>

* chore(agw): Clean up unnecessary Make commands: log, logs and status

Signed-off-by: Lars Kreutzer <lars.kreutzer@tngtech.com>

* chore(agw): Clean up unnecessary Make commands: precommit_sm and precommit_oai

Signed-off-by: Lars Kreutzer <lars.kreutzer@tngtech.com>

* chore(agw): Clean up unnecessary Make commands: scan_oai

Signed-off-by: Lars Kreutzer <lars.kreutzer@tngtech.com>

* chore(agw): Clean up unnecessary Make commands: smf_build

Signed-off-by: Lars Kreutzer <lars.kreutzer@tngtech.com>

Signed-off-by: Lars Kreutzer <lars.kreutzer@tngtech.com>

rsarwad committed:
fix(agw): Migrate hash list of state_ue to protobuf_map (#13508)

Modified code to replace hashtable with protobuf map for state_ue hash table in s1ap module

sebathomas committed:
fix(agw): Fix spyne version in pydep lockfile (#13809)

Signed-off-by: Sebastian Thomas <sebastian.thomas@tngtech.com>

LKreutzer committed:
fix(ci): Set up the scripts needed in the bazel LTE integ test workflow (#13803)

Signed-off-by: Lars Kreutzer <lars.kreutzer@tngtech.com>
Co-authored-by: Lars Kreutzer lars.kreutzer@tngtech.com
Co-authored-by: Krisztián Varga krisztian.varga@tngtech.com

Signed-off-by: Lars Kreutzer <lars.kreutzer@tngtech.com>

Commit:
fix(ci): Fix feg integ tests. Feg integ tests need both docker magmad for feg and systemd magmad for agw. So the detection fails. (#13781)

Signed-off-by: Christian Krämer <christian.kraemer@tngtech.com>

Signed-off-by: Christian Krämer <christian.kraemer@tngtech.com>

LKreutzer committed:
fix(ci): Trigger all bazel workflow jobs by pushing to master (#13825)

Signed-off-by: Lars Kreutzer <lars.kreutzer@tngtech.com>

Signed-off-by: Lars Kreutzer <lars.kreutzer@tngtech.com>

Commit:
fix(ci): fix ipv6 uplink test (#13669)

* fix(agw): add correct address prefix for the local ue
* fix(agw): Make multicast solicitation packages pass to the local dev vm interface in order to make the agw reachable in ipv6 bridged mode
* fix(ci): reenable ipv6 upload test

Signed-off-by: Christian Krämer <christian.kraemer@tngtech.com>

ssanadhya reviewed (approved):
Lgtm
